packages/build/src/plugins/spawn.ts

```diff
@@ -45,7 +47,17 @@ const tStartPlugins = async function ({ pluginsOptions, buildDir, childEnv, logs
 export const startPlugins = measureDuration(tStartPlugins, 'start_plugins')
 
 const startPlugin = async function ({ pluginDir, nodePath, buildDir, childEnv, systemLogFile, pluginPackageJson }) {
-  const childProcess = execaNode(CHILD_MAIN_FILE, [], {
+  const ctx = trace.getActiveSpan()?.spanContext()
+
+  // the baggage will be passed to the child process when sending the run event
+  const args = [
+    ...process.argv.filter((arg) => arg.startsWith('--tracing')),
+    `--tracing.traceId=${ctx?.traceId}`,
+    `--tracing.parentSpanId=${ctx?.spanId}`,
+    `--tracing.traceFlags=${ctx?.traceFlags}`,
+  ]
+
+  const childProcess = execaNode(CHILD_MAIN_FILE, args, {
```

Comment:
This will be passed to all plugins - is that ok or do we maybe want to limit this to plugins we own? (consider `isTrustedPlugin` check possibly)

And if we want to limit which plugins we enable tracing support on - possibly other places might need conditional execution as well then

Reply:
Done in https://github.com/netlify/build/pull/5556/commits/58ea6746ad92df43d6b76f517e675de7b9121827